src/test/java/com/exactpro/th2/crawler/CrawlerTest.java

```diff
@@ -306,4 +313,135 @@ public void dataServiceFail() throws IOException, UnexpectedDataProcessorExcepti
+                from, to, DataType.MESSAGES, Set.of(CrawlerManager.SESSIONS));
+
+        CrawlerManager manager = new CrawlerManager(configuration);
+        Crawler crawler = manager.createCrawler();
+
+        StateService<RecoveryState> stateService =
+                StateService.createFromClasspath(RecoveryState.class, Mockito.mock(DataProviderService.class), null);
+
+        var state = createRecoveryState();
+
+        Collection<Message> messages = MessageReaderKt.readMessages(Paths.get("src/test/resources/messages.txt"));
+        Interval interval = new IntervalBuilder()
+                .crawlerName("name")
+                .crawlerVersion("v1")
+                .crawlerType("MESSAGES")
+                .startTime(Instant.parse(from))
+                .endTime(Instant.parse(intervalEnd))
+                .processed(false)
+                .recoveryState(stateService.serialize(state))
+                .build();
+
+        manager.getIntervalsWorkerMock().storeInterval(interval);
+
+        Iterator<StreamResponse> iterator = new MessageSearchResponse(messages).iterator();
+        when(manager.getDataProviderMock().searchMessages(any(MessageSearchRequest.class))).thenReturn(iterator);
+        when(manager.getDataServiceMock().sendMessage(any(MessageDataRequest.class))).thenReturn(MessageResponse.getDefaultInstance());
+
+        crawler.process();
+
+        Assertions.assertTrue(interval.isProcessed());
+    }
+
+    @Test
+    @DisplayName("Crawler takes unprocessed interval in MULTIPLE mode with FIXED endTime")
+    public void unprocessedIntervalMultipleModeFixedEnd() throws IOException, UnexpectedDataProcessorException {
+        String from = "2021-06-16T12:00:00.00Z";
+        String to = "2021-06-16T20:00:00.00Z";
+        String intervalEnd = "2021-06-16T13:00:00.00Z";
+        String lastUpdateTime = "2021-06-16T10:00:00.00Z";
+
+        CrawlerConfiguration configuration = new CrawlerConfiguration(
+                from, to, "test_crawler", DataType.MESSAGES, "PT1H", 1, ChronoUnit.NANOS,
+                1, 10, 5, ChronoUnit.MINUTES, false,
+                Set.of(CrawlerManager.SESSIONS), GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE);
+
+        CrawlerManager manager = new CrawlerManager(configuration);
+
+
+        Crawler crawler = manager.createCrawler();
+
+        StateService<RecoveryState> stateService =
+                StateService.createFromClasspath(RecoveryState.class, Mockito.mock(DataProviderService.class), null);
+
+        var state = createRecoveryState();
+
+        Collection<Message> messages = MessageReaderKt.readMessages(Paths.get("src/test/resources/messages.txt"));
+        Interval interval = new IntervalBuilder()
+                .crawlerName("name")
+                .crawlerVersion("v1")
+                .crawlerType("MESSAGES")
+                .startTime(Instant.parse(from))
+                .endTime(Instant.parse(intervalEnd))
+                .processed(false)
```

Comment:
Could you please add an additional interval before existing that one has the last update interval less than the last update check option?
We expected the crawler chooses
* the additional interval in work alone mode
* the exited interval in work multiple mode